crates/sui-types/src/gas_model/tables.rs

```diff
@@ -76,7 +75,6 @@ pub struct GasStatus {
     instructions_next_tier_start: Option<u64>,
     instructions_current_tier_mult: u64,
 
-    #[cfg(debug_assertions)]
     profiler: Option<GasProfiler>,
```

Comment:
Related to my earlier comment, this really implies that `GasProfiler` is always available, under all configurations -- otherwise this would not successfully compile -- but this is not what we want.

Reply:
When people say that features need to be additive, they mean that the code needs to be such that I can (not necessarily will) have any combination of features turned on. Here we had declared the trait in crate A with set / get functions that we attempt to feature gate. Then we used that trait in other crates B, C, D, each that define their own feature to gate the respected implementations. But if the feature is turned on in only crate A, or in only in B, C, or D and not A, this causes code that cannot be build which is not correct usage of features.

Reply:
The property of features being additive is local to a crate, i.e. every crate should build with any subset of its supported features enabled.  It should be possible to gate the inclusion of the gas profiler into the gas status by a feature in the `sui-types` crate which also controls whether `move-vm-profiler` includes the `GasProfiler` type.

Reply:
After fixing the bugs I tried to add back feature gating to profiler as struct attribute, but again I ran into build failures that I saw earlier, due to lack of common ancestor